src/process/volcanic/CCPr_Volcanic_Mod.F90

```diff
@@ -0,0 +1,381 @@
+
+      else
+         VolcanicState%Activate = .false.
+      end if
+
+   end subroutine CCPR_Volcanic_Init
+
+   !>
+   !! \brief Run the VolcanicEmissions
+   !!
+   !! \param [IN] MetState - The MetState object
+   !! \param [INOUT] VolcanicState - The VolcanicState object
+   !! \param [INOUT] EmisState - The EmisState object
+   !! \param [INOUT] RC Return code
+   !!
+   !! \ingroup catchem_Volcanicemissions_process
+   !!!>
+   SUBROUTINE CCPr_Volcanic_Run( MetState, VolcanicState, EmisState, RC )
+
+      ! USE
+      USE constants, only : g0
+      use CCPr_Scheme_Volcanic_GOCART_Mod, only : CCPr_Scheme_Volcanic_GOCART, VolcanicEmisData, ReadASCIIPointEmis
+
+      IMPLICIT NONE
+      ! INPUT PARAMETERS
+      TYPE(MetStateType),  INTENT(IN) :: MetState       !< MetState Instance
+
+      ! INPUT/OUTPUT PARAMETERS
+      !TYPE(DiagStateType), INTENT(INOUT)      :: DiagState       !< DiagState Instance
+      TYPE(VolcanicStateType), INTENT(INOUT)  :: VolcanicState  !< VolcanicState Instance
+      !TYPE(ChemStateType), INTENT(INOUT)     :: ChemState       !< ChemState Instance
+      TYPE(EmisStateType), INTENT(INOUT)     :: EmisState       !< ChemState Instance
+
+      ! OUTPUT PARAMETERS
+      INTEGER, INTENT(INOUT) :: RC                                 ! Return Code
+
+      ! LOCAL VARIABLES
+      CHARACTER(LEN=255) :: ErrMsg, thisLoc
+      CHARACTER(LEN=1055) :: fname
+      type(VolcanicEmisData), allocatable :: VolcanicEmis(:)
+      CHARACTER(len=7), parameter :: label='volcano'
+      INTEGER :: i                          ! loop index
+      INTEGER :: hms                        ! Model time [secs] TODO: format is right?
+      INTEGER :: ymd                        ! Model date [YYYYMMDD] TODO: format is right?
+      CHARACTER(LEN=18) :: ymd_str          ! Model date in string
+
+      INTEGER, dimension(:), allocatable   :: vStart      ! Emissions Start time [sec]
+      INTEGER, dimension(:), allocatable   :: vEnd        ! Emissions end time [sec]
+      INTEGER                              :: nVolc       ! number of volcanic sources
+      INTEGER, dimension(:), allocatable   :: iPoint, jPoint ! sub-domain - we only run this at the place/time of eruption??
+      INTEGER, parameter                   :: nSO2 =1     ! index of SO2 relative to other sulfate tracers
+
+      REAL, dimension(:), allocatable      :: vSO2   ! volcanic emissions  [kg]
+      REAL, dimension(:,:,:),pointer  :: SO2       ! SO2 [kg kg-1]
+      !REAL, dimension(:,:,:),pointer  :: SU_emis   ! SU emissions, kg/m2/s
+      REAL, dimension(:), allocatable        :: vCloud    ! top elevation of emissions [m]
+      REAL, dimension(:), allocatable        :: vElev     ! bottom elevation of emissions [m]
+      REAL, dimension(:), allocatable        :: vLat     ! latitude specified in file [degree]
+      REAL, dimension(:), allocatable        :: VLon     ! longitude specified in file [degree]
```

Comment:
It looks like this should really be vLon to be consistent with that below and the other tracers.